### ISSUE TITLE
Building MSI: Legal version values should look like 'x.x.x.x'

### ISSUE THREAD

**Issue by remco1271** [clowd/Clowd.Squirrel#59]
Building the release fails on the MSI build.
The -delta.nupkg file is build without a problem.
My exe has SquirrelAwareVersion enabled.
Copy CanBus.exe to CAN.exe doesn't change the error

### Exe info:

- name: CanBus.exe
- version: 1.0.16
- .net version: net6.0

### Console log:
```
[INFO] Program: Creating Setup bundle
[INFO] Program: Compiling machine-wide msi deployment tool in 64-bit mode
[WARN] Utility: Cannot delete 'C:\Users\remco1271\source\repos\can\Releases\CANSetup.wixobj' if it does not exist.


[ERRO] System.AggregateException: One or more errors occurred. (Failed to compile WiX template, command invoked was: 'candle.exe System.String[]'

Output was:
CANSetup.wxs
C:\Users\remco1271\source\repos\can\Releases\CANSetup.wxs(3) : error CNDL0108 : The Product/@Version attribute's value, '!(bind.FileVersion.CAN.exe)', is not a valid version.  Legal version values should look like 'x.x.x.x' where x is an integer from 0 to
 65534.
C:\Users\remco1271\source\repos\can\Releases\CANSetup.wxs(3) : error CNDL0010 : The Product/@Version attribute was not found; it is required.)
 ---> System.Exception: Failed to compile WiX template, command invoked was: 'candle.exe System.String[]'

Output was:
CANSetup.wxs
C:\Users\remco1271\source\repos\can\Releases\CANSetup.wxs(3) : error CNDL0108 : The Product/@Version attribute's value, '!(bind.FileVersion.CAN.exe)', is not a valid version.  Legal version values should look like 'x.x.x.x' where x is an integer from 0 to
 65534.
C:\Users\remco1271\source\repos\can\Releases\CANSetup.wxs(3) : error CNDL0010 : The Product/@Version attribute was not found; it is required.
   at Squirrel.HelperExe.CompileWixTemplateToMsi(String wxsTarget, String outputFile) in ./Internal/HelperExe.cs:line 102
   at SquirrelCli.Program.createMsiPackage(String setupExe, IPackage package, Boolean packageAs64Bit) in ./Program.cs:line 371
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean )
   at SquirrelCli.Program.Releasify(ReleasifyOptions options)
   at SquirrelCli.Program.Pack(PackOptions options) in ./Program.cs:line 127
   at SquirrelCli.CommandAction`1.Execute(IEnumerable`1 args) in ./ValidatedOptionSet.cs:line 160
   at SquirrelCli.CommandSet.Execute(String[] args) in ./ValidatedOptionSet.cs:line 194
   at SquirrelCli.Program.Main(String[] args) in ./Program.cs:line 80



Squirrel (2.8.15-pre+dfc5b5) command line tool for creating and deploying Squirrel releases
Usage: Squirrel.exe [verb] [--option:value]
```

**Comment by caesay** [comment]
The string `!(bind.FileVersion.CAN.exe)` from the error is a little misleading, because it's really looking for `!(bind.FileVersion.{Id})` and the `{Id}` just happens to be `{packageName}.exe`. The actual name of your files is not important.

I have noticed that, when googling for this issue, lots of people on the original Squirrel also ran into this error for various reasons. (eg. https://github.com/electron/windows-installer/issues/187)

I have refactored the WiX template in 11e471fdadd198546b491093ef9f5be3eadf4bac to no longer use this WiX binding syntax at all. This is released [in 2.8.24-pre](https://github.com/clowd/Clowd.Squirrel/releases/tag/2.8.24-pre). This should also fix the related electron issues, if this library ever gets adopted over there.

Please let me know if this resolves your issue!

**Comment by remco1271** [comment]
The update did not change it.
Did solve it by changing the packId to the name of the ".exe" file.
Could you not check that the {packageName}.exe exist before try to build the msi?

**Comment by caesay** [comment]
What is the new error please? I would not have expected the packId to matter here, and I've tested this with random packId's to verify on my end.

**Comment by caesay** [comment]
Also if you could please provide me with the command line that was passed into Squirrel, that would be super helpful!

**Comment by remco1271** [comment]
I do use powershell to find the lastest Squirel
The "-" in the packId is the issue if I remove it it works
```
$core = ls -Filter * ($env:USERPROFILE + "\.nuget\packages\clowd.squirrel\") | sort { [version]($_.Name -replace '^(\d+(\.\d+){1,3}).*$', '$1') } -Descending | select -Index 0
Set-Alias Squirrel ($core.FullName + "\tools\Squirrel.exe");
echo "Using Squirrel version: $core.Name"
Squirrel pack `
    --framework net6 `
    --packId "Can-Bus" `
    --packTitle "CAN-Bus monitor" `
    --packVersion "$verison" `
    --packAuthors "Remco" `
    --packDirectory ".\CanBus\bin\Publish" `
    --icon ".\Icons\ico\CAN.ico" `
    --msi "x64" `
    --splashImage ".\splash.gif"
```

**Comment by caesay** [comment]
Thanks for the additional details here! I have been able to reproduce this. I fixed a lot of bugs with dots and dashes in package names a while ago but I guess I missed a few related to WiX as I don't use this feature myself. Will post a patch asap!

**Comment by caesay** [comment]
This has been fixed and released [in 2.8.30-pre.](https://github.com/clowd/Clowd.Squirrel/releases/tag/2.8.30-pre)
